qualtran/bloqs/block_encoding.py

```diff
@@ -182,13 +186,46 @@ def pretty_name(self) -> str:
         return 'Prep'
 
 
+class BlockEncoding(Bloq):
+    r"""Abstract interface for an arbitrary block encoding.
+
+    Users must provide methods for defining the ancilla, junk, and system registers,
+    and a method for defining the signal state $|G\rangle$. See module level
+    docstrings for more discussion.
+
+    References:
+        [Hamiltonian Simulation by Qubitization](https://quantum-journal.org/papers/q-2019-07-12-163/)
+            Sec 2 and 3 for introduction and definition of terms.
+    """
+
+    @property
+    @abc.abstractmethod
+    def selection_bitsize(self) -> int:
+        """The selection bitsize for state preparation. These are the `a` registers above"""
+
+    @property
+    @abc.abstractmethod
+    def junk_bitsize(self) -> int:
+        """The junk bitsize (additional ancilla registers not reflected on)."""
+
+    @property
+    @abc.abstractmethod
+    def system_bitsize(self) -> int:
+        """The system bitsize for applying the controlled (signal) unitary to."""
+
+    @property
+    @abc.abstractmethod
+    def reflection_using_signal_state(self):
```

Comment:
I think this should just return a Bloq that prepares the initial state instead of providing a reflection around it? The "reflection around signal state"  is a feature of qubitization that doesn't necessarily need to be part of the block encoding base class.

Reply:
It's called signal state in the qubitization reference, but I'm not wedded to the name. Do you know if there's something better? Yeah, I originally just had signal_state, but it seemed more convnenient for qubitization but you're right, it's something we can push to the walk operator to just get $|G\rangle$ from B to prepare the reflection.